Tests/UnitTests/CompletionHandlerTests.swift

```diff
@@ -0,0 +1,190 @@
+//
+// Copyright 2020 Adobe. All rights reserved.
+// This file is licensed to you under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may obtain a copy
+// of the License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under
+// the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+// OF ANY KIND, either express or implied. See the License for the specific language
+// governing permissions and limitations under the License.
+//
+
+import AEPCore
+@testable import AEPEdge
+import XCTest
+
+class CompletionHandlerTests: XCTestCase {
+    private let requestEventId = "requestEventId"
+    // swiftlint:disable force_try
+    private var eventHandle: EdgeEventHandle = {
+        let json: [String: Any] = ["payload": [["key1": "value1", "key2": 2]], "type": "testType"]
+        let jsonData = try! JSONSerialization.data(withJSONObject: json, options: [])
+        return try! JSONDecoder().decode(EdgeEventHandle.self, from: jsonData)
+    }()
+    private var eventError: EdgeEventError = {
+        let json: [String: Any] = ["message": "Error message", "code": "789", "namespace": "test.namespace"]
+        let jsonData = try! JSONSerialization.data(withJSONObject: json, options: [])
+        return try! JSONDecoder().decode(EdgeEventError.self, from: jsonData)
+    }()
+    // swiftlint:enable force_try
+    private let uniqueEventId = "123"
+
+    override func setUp() {
+        continueAfterFailure = false // fail so nil checks stop execution
+    }
+
+    func testRegisterCompletionHandler_thenEventHandlerReceived_completionCalled() {
```

Comment:
In these tests, we should also set `assertForOverfullfill` to ensure they are only invoked once.
https://developer.apple.com/documentation/xctest/xctestexpectation/2806575-assertforoverfulfill